src/tools/apollo-tool/apollo-tuner.cpp

```diff
@@ -1,142 +1,179 @@
 
+#include <algorithm>
+#include <apollo/Apollo.h>
+#include <apollo/Region.h>
 #include <cassert>
+#include <chrono>
 #include <cinttypes>
 #include <cstdio>
-#include <algorithm>
 #include <cstring>
-#include <utility>
 #include <functional>
+#include <impl/Kokkos_Profiling_Interface.hpp>
+#include <impl/Kokkos_Tuning_Interface.hpp>
 #include <map>
 #include <queue>
 #include <set>
 #include <string>
-#include <vector>
-#include <chrono>
 #include <tuple>
-#include <impl/Kokkos_Profiling_Interface.hpp>
-#include <impl/Kokkos_Tuning_Interface.hpp>
-#include <apollo/Apollo.h>
-#include <apollo/Region.h>
-#define TUNING_TOOL_SAVE_STATISTICS  // TODO DZP make build system option
+#include <utility>
+#include <vector>
+#define TUNING_TOOL_SAVE_STATISTICS // TODO DZP make build system option
 
 std::map<std::string, Apollo::Region> regions;
 std::set<size_t> untunables;
 std::set<size_t> unlearnables;
 std::string activeRegion = "";
 constexpr const int max_choices = 2 << 20;
+constexpr const int max_variables = 50;
+
+struct variableSet {
+  size_t variable_ids[max_variables];
+  size_t num_variables;
+};
 
 int choices[max_choices];
 
-Apollo* apollo; 
+namespace std {
+template <> struct less<variableSet> {
+  bool operator()(const variableSet &l, const variableSet &r) {
```

Comment:
I'm curious why you would want to specialize `std::less`, rather than just define `operator<` for your type.  Unspecialized `std::less` just calls `operator<`.

Reply:
No conscious reason, I'm not too familiar with achieving compatibility with `std`-namespaced containers and the code I'm aping using this pattern

Reply:
It's both easier to define `operator<`, and more in line with developers' expectations.  If you do that, you'll get the `std::less` specialization for free.  I would find it weird to have a `std::less` specialization, but not have `operator<` work.

Reply:
If that's what's standard I'll probably implement it when it comes time to merge. 

For now I'm not sure whether I might not just tear this all down and do something completely different, but it's a good comment if this makes it the distance. Thanks!